[HERETICS] Void & a handful of heretics-related PRS

kiwedespars opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

ports void path for heretics, and a ton of other prs related to heretics from tg, along with a few additions/changes.

infinite void - after a 5 second windup, make a 7x7 area your temporary domain, slowing enemies inside and also marking them with a void mark.

((voids repulse (placeholder) - snapping fingers sends anything nearby flying back.)) goodbye.

void's applause - clap your hands and switch places with someone of your choice.

credits to 
tgstation/tgstation#54252
tgstation/tgstation#54371
tgstation/tgstation#54705
tgstation/tgstation#55050
tgstation/tgstation#55437
tgstation/tgstation#55630
tgstation/tgstation#55667
tgstation/tgstation#56268
tgstation/tgstation#56304
tgstation/tgstation#56441

## Why It's Good For The Game

more antag content good.

## Changelog
:cl:
add: New heretic path - Path of Void- it specializes in being extremely stealthy.
tweak : Removed curse of blindness replaced with mask of madness. hey it even rhymes.
fix: fixes heretic mass deletion during transmutation bug.
fix: Fixes heretic brews being permanent.
fix: Fixes void storm breaking after resurrecting.
tweak: Heretic has received a minor textual facelift.
tweak: Heretics who finish the Void Path and become an Aristocrat of the Void can now survive in the Void (space).
fix: Heretics who research Aristocrats Way on the Void Path will no longer suffocate in their own storm when they ascend (no longer breathes).
fix: Mark of Void and Seeking Blade are once again exclusive only to Void.
balance: Carving knife now deals more damage on throw and can embed in your enemies.
tweak: Grasp of Rust only rusts floors and machines on harm intent.
tweak: when u choose a sac target as heretic it ll also tell the job of the sac
add: 2 new void spells, one a placeholder.
add: flesh mansus grasp buffed, now gives you 5u eldritch fluid when hitting someone.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

kiwedespars committed:
Merge remote-tracking branch 'upstream/master'

kiwedespars committed:
Update rust_g.dll

kiwedespars committed:
void heretic and many other prs

kiwedespars committed:
exist

- fixes things.
- user check readded (whoops)
- knwoledge added
- nerfs madness mask

kiwedespars committed:
Merge remote-tracking branch 'upstream/master'

kiwedespars committed:
Merge branch 'master' into void-but-for-real

kiwedespars committed:
whoops forgot sac targets list

kiwedespars committed:
trailing newlines

actions-user committed:
Automatic changelog compile [ci skip]

kiwedespars committed:
Update rust_g.dll

kiwedespars committed:
more influences

kiwedespars committed:
goodbye eldritch blade

kiwedespars committed:
void tp effect, sound changes

kiwedespars committed:
makes void blade a little easier to get

kiwedespars committed:
i accidentally overwrote this, fixes logging.

kiwedespars committed:
Merge remote-tracking branch 'upstream/master'

actions-user committed:
Automatic changelog compile [ci skip]

kiwedespars committed:
adds domain expansion

- adds domain expansion, final spell for void heretics
- makes clap1 into an emote
- makes cleave a little more enticing with two guaranteed critical wounds and brute damage
- buffs void mark

kiwedespars committed:
brain is like monkey, it forgets the newlines

kiwedespars committed:
better sound

kiwedespars committed:
Merge branch 'master' into void-but-for-real

kiwedespars committed:
Merge remote-tracking branch 'upstream/master'

kiwedespars committed:
Update rust_g.dll

kiwedespars committed:
Merge branch 'master' into void-but-for-real

kiwedespars committed:
did this not update properly

kiwedespars committed:
fixes marks

- marks are no longer set by the blade and detonated by the hand, outside of flesh
- fixed it so sickles can no longer be used by nonheretics, again, needs testing for heretic minions (ghouls)

kiwedespars committed:
readded succumb button and also removes halo.

kiwedespars committed:
nerf sickle break

necromanceranne commented on code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm:
```diff
@@ -0,0 +1,223 @@
+	for(var/mob/living/carbon/livies in spiral_range(7,user)-user)
+		if(IS_HERETIC_MONSTER(livies) || IS_HERETIC(livies))
+			return
+		livies.silent += 1
```
Put a clamp on this.

necromanceranne commented on code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm:
```diff
@@ -0,0 +1,223 @@
+	if(!E)
+		return
+	E.on_effect()
+	H.silent += 3
```
Here too if you would.

necromanceranne commented on code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm:
```diff
@@ -0,0 +1,223 @@
+	var/turf/open/turfie = get_turf(carbon_target)
+	turfie.TakeTemperature(-20)
+	carbon_target.adjust_bodytemperature(-40)
+	carbon_target.silent += 4
```
Clampies

kiwedespars committed:
changes

- void blade requires colder temps
- boogie woogie fueltank swap
- sicklebreak buff
- big domain expansion buff
- grasping carving nerf
- clamps silence maybe

kiwedespars committed:
maybe too big a buff
